pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -851,6 +858,20 @@ func configFromProviderSpec(providerSpec machinev1beta1.ProviderSpec) (*bmv1alph
 	return &config, nil
 }
 
+// clearMachineAddresses clears the Host IP addresses in the Machine status, so
+// that another Host using the same IP can later be used by another Machine.
+func (a *Actuator) clearMachineAddresses(ctx context.Context, machine *machinev1beta1.Machine) error {
+	var err error
+	if machine.ObjectMeta.DeletionTimestamp.IsZero() && len(machine.Status.Addresses) > 0 {
```

Comment:
It's correct - Exists() gets called both in the Provisioned/Running phases (before Update()) and in the Deleting phase (after Delete()), and we only want to clear the addresses in the former case (in the latter case, there's no problem because the Machine is going away).

Reply:
That may deserve a comment, too. Keeping these cases straight with the actuator API is non-trivial.